src/button/borderless.scss

```diff
@@ -2,65 +2,77 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 @import './button';
+@import './disabled';
+@import './button-icon';
 
 @mixin iui-button-borderless {
-  background-color: transparent;
-  border-color: transparent;
-  padding: 0 $iui-s;
-
-  // Sizes
-  &.iui-small {
-    padding: 0 $iui-xxs;
-  }
-
-  &.iui-large {
-    padding: 0 $iui-sm;
-  }
+  @include iui-button-borderless-base;
 
-  // If icon is by itself, button takes 1:1 size ratio
-  > .iui-icon:only-child {
-    margin-left: $iui-component-offset;
-    margin-right: $iui-component-offset;
-  }
-
-  &:enabled:hover,
-  &:enabled:active,
-  &:focus-visible {
+  &:hover,
+  &:focus {
     @include themed {
       background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-6));
       border-color: transparent;
     }
+
+    > .iui-button-icon {
+      @include iui-button-icon(t(iui-icons-color-actionable-hover));
+    }
   }
 
-  &.iui-active:not(.iui-high-visibility):not(.iui-cta) {
-    &:enabled,
-    &:enabled:hover,
-    &:enabled:active {
+  &:focus:not(:focus-visible) {
+    @include iui-button-borderless-base;
+  }
+
+  &.iui-active {
+    &,
+    &:hover {
       background: transparent;
       @include themed {
         background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
       }
       border-color: transparent;
+
+      > .iui-button-label {
+        @include themed {
+          color: t(iui-color-foreground-primary);
+        }
+      }
+
+      > .iui-button-icon {
+        @include iui-button-icon(t(iui-icons-color-primary));
+      }
     }
   }
 
+  &[disabled] {
+    @include iui-button-disabled;
+  }
+
   &[disabled],
-  &[disabled]:hover,
-  &[disabled]:active,
-  &[disabled]:focus {
+  &:disabled,
+  &:disabled:hover {
```

Comment:
This selector is messed up.

1. Why have `[disabled]` separately for including the mixin?
2. Why include `hover` here?